NEWS.md

```diff
@@ -24,6 +24,8 @@ Command-line option changes
 
 * In Linux and Windows, `--threads=auto` now tries to infer usable number of CPUs from the
   process affinity which is set typically in HPC and cloud environments ([#42340]).
+* `--math-mode=fast` is now a no-op ([#41638]). local fastmath annotations (eg `@fastmath`)
+  give the same performance without breaking your code as much.
```

Comment:
The statement "without breaking your code as much" sounds a bit cryptic and confusing. My suggestion is to either explain better or not refer to breakage and let people read the linked issue/PR. This reads conversational and casual and not like release notes.

Reply:
Perhaps replace the second sentence with "Users are encouraged to use the `@fastmath` macro instead, which has more well-defined semantics."?